src/traces/parcats/attributes.js

```diff
@@ -0,0 +1,182 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var extendFlat = require('../../lib/extend').extendFlat;
+var colorAttributes = require('../../components/colorscale/attributes');
+
+var scatterAttrs = require('../scatter/attributes');
+var scatterMarkerAttrs = scatterAttrs.marker;
+var colorbarAttrs = require('../../components/colorbar/attributes');
+
+var marker = extendFlat({
+    editType: 'calc'
+}, colorAttributes('marker', {editType: 'calc'}),
```

Comment:
If we at some point want to support selections similar to how parcoords works, perhaps we want [Viridis](https://github.com/plotly/plotly.js/blob/0c1d6d5ebbd2334f712aede2a32236fb451838b7/src/traces/parcoords/attributes.js#L119-L121) here too? How well does this work with continuous color values, does it attempt to sort in color order or does it just keep distinct values together? Also I don't see a mock with a colorbar and in your brushing example it looks like you just use two distinct colors - does it support providing an array of color strings or just numeric values? Actually I'm not sure if `parcoords` supports strings...

Anyway if you're just using color as a marker of selected/deselected it seems a bit hacky to have to create a colorscale with the two colors you want. Perhaps instead (or in addition) we should use [`selectedpoints`](https://github.com/plotly/plotly.js/blob/master/src/plots/attributes.js#L104) and add [`selected`/`unselected` styling](https://github.com/plotly/plotly.js/blob/0c1d6d5ebbd2334f712aede2a32236fb451838b7/src/traces/scatter/attributes.js#L369-L440), with the default for `unselected` being a grey similar to `parcoords` and your brushing example.

Reply:
A path (thick line) is created for each unique combination of the dimension categories and raw color (the value as specified in `marker.color`).  The relative height of the path is determined by the number of observations the path represents.

The display color of the path is computed based on the path's unique raw color value. If no colorscale is present, then the raw color is the display color (So yes, you can pass an array of color strings to `marker.color`).  If a colorscale is present and the raw color is numeric then it is mapped to a display color using the usual scaling process.

So continuous colorscales are fine as long as the numeric color values are somewhat discrete.  If there are 1000 unique values in the color array, then the diagram will sort of degenerate into a bunch of tiny paths.  In my own analysis work I've either used a categorical colormap with an integer color array (this is the brushing scenario), or used a continuous colormap and manually discretized the color values. In both cases I would generally stay under 10 discrete colors.

The selection metaphor here is interesting. So far I'm just emitting the `plotly_click` and hover events and then executing selection like logic (highlighting clicked paths) from the outside.